session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java

```diff
@@ -76,6 +76,8 @@
   private boolean enableCompression = false;
   private ZoneId zoneId;
 
+  private boolean closed;//whether the queue is closed.
```

Comment:
I think it should be volatile, cause it is used outside a synchronized block somewhere below

Reply:
but it does not take native effective if a thread sees the value of `closed` with a latency.